Automattic/wp-calypso
# Components: Add Jetpack Upsell component skeleton

First PR for #71820.

## Proposed Changes
Design|Implementation
-|-
<img width="1248" alt="image" src="https://user-images.githubusercontent.com/6406071/213558155-f7dad638-85d7-471d-9f40-a49621bdd8bf.png">|<img width="1024" alt="image" src="https://user-images.githubusercontent.com/4044428/219220868-902bd85b-360f-47f9-9e22-d7ba6f0f5f27.png">

Adds a Jetpack Upsell component to the components package.

## Testing Instructions



* Open a Storybook instance for `@automattic/components` via `yarn workspace @automattic/components run storybook`.
* Ensure you can see a new "Jetpack Upsell Section" component for preview. Note that SF Pro fonts do not load in Storybook, so the fonts may not look as specified by design.

## Pre-merge Checklist
